app/controllers/api/deploy_environments_controller.rb

```diff
@@ -59,6 +59,20 @@ def validate_config_file
     system({}, "docker run --rm -v /tmp:/tmp #{current_environment.repository}:#{params["version"]} sh -c 'rm #{schema_file_tmp_path}'")
   end
 
+  def create
+     p "triggered deploy_environment api"
+     deploy_environment = current_user.deploy_environments.find(params[:pr_deployment_environment_id])
+     temp_environment_name = deploy_environment[:name].split("-")
+     temp_environment_name[temp_environment_name.length - 1] = "pr" + params[:pr_num].to_s
+     new_deploy_environment_name = temp_environment_name.join("-")
+
+     new_deploy_environment              = deploy_environment.amoeba_dup
+     new_deploy_environment.name         = new_deploy_environment_name
+     new_deploy_environment.app_name     = new_deploy_environment_name
+     new_deploy_environment.publisher_id = params[:pr_publisher_id]
```

Comment:
You can do `new_deploy_environment.assign_attributes(name: new_deploy_environment_name, app_name: new_deploy_environment_name, publisher_id: params[:pr_publisher_id] )`